pkg/workloadslicing/workloadslicing.go

```diff
@@ -104,17 +104,20 @@ func Finish(ctx context.Context, clnt client.Client, clk clock.Clock, workloadSl
 	if apimeta.IsStatusConditionTrue(workloadSlice.Status.Conditions, kueue.WorkloadFinished) {
 		return nil
 	}
+	now := clk.Now()
 	if err := clientutil.PatchStatus(ctx, clnt, workloadSlice, func() (client.Object, bool, error) {
 		return workloadSlice, apimeta.SetStatusCondition(&workloadSlice.Status.Conditions, metav1.Condition{
 			Type:               kueue.WorkloadFinished,
 			Status:             metav1.ConditionTrue,
 			Reason:             reason,
 			Message:            message,
-			LastTransitionTime: metav1.NewTime(clk.Now()),
+			LastTransitionTime: metav1.NewTime(now),
 		}), nil
 	}); err != nil {
 		return fmt.Errorf("failed to patch workload slice status: %w", err)
 	}
+
+	workload.ReportEvictionCompleted(workloadSlice, kueue.WorkloadFinished, now)
```

Comment:
In this context, Finished does not necessarily (or always) implies "Eviction". 
We probably shouldn't track eviction duration here.

Reply:
ReportEvictionCompleted calls: https://github.com/kubernetes-sigs/kueue/pull/6477/files#diff-a7490e7f5efa2a59ab508f592b67dd6e7530b9d2e1b19a1b6a7438e6c2386905R1359

Here Eviction duration is not emitted unless it sees an eviction condition.